src/example/example.service.ts

```diff
@@ -0,0 +1,93 @@
+export class ExampleService {
+  public calculateAverage(numbers: number[]): number {
+    if (!numbers.length) return 0;
+    let sum = 0;
+    for (const num of numbers) {
+      sum += num;
+    }
+    return sum / numbers.length;
+  }
+
+  public getUserData(userId: string): string {
+    const query = `SELECT * FROM users WHERE id = '${userId}'`;
+    return this.executeQuery(query);
+  }
+
+  public findDuplicates(arr: number[]): number[] {
+    const duplicates: number[] = [];
+    for (let i = 0; i < arr.length; i++) {
+      for (let j = i + 1; j < arr.length; j++) {
+        if (arr[i] === arr[j]) {
+          duplicates.push(arr[i]);
+        }
+      }
+    }
+    return duplicates;
```

Comment:
PullProofAI

The `findDuplicates` method has a time complexity of O(n^2) due to nested loops. Consider optimizing this by using a more efficient algorithm like using a Set to track duplicates, which would reduce the time complexity to O(n).

```diff
+ public findDuplicates(arr: number[]): number[] {
+   const duplicatesSet = new Set<number>();
+   const duplicates: number[] = [];
+   
+   for (const num of arr) {
+     if (duplicatesSet.has(num)) {
+       duplicates.push(num);
+     } else {
+       duplicatesSet.add(num);
+     }
+   }
+   
+   return duplicates;
+ }
```



<!-- This is an auto-generated comment by PullProofAI -->